### PR TITLE
fix: not working on Windows 10

### LINKED ISSUE #20: Not working on Windows 10
I got an error on my Windows 10 computer when I tried to run eslint.

``` sh
PS C:\path\to\repository> npx eslint-cjs-to-esm "./src/**/*.{js,ts}"
C:\path\to\repository\node_modules\.bin\eslint:2
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
          ^^^^^^^

SyntaxError: missing ) after argument list
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49

Node.js v20.11.0
```

An error is occurring in the execution of the eslint sh file.

https://github.com/azu/eslint-cjs-to-esm/blob/main/eslint-cjs-to-esm.js#L6

I am not familiar with how eslint or npm bin works, but it seems that on Windows bat files are used instead of sh files.

image

Similar Issues: 

- https://github.com/gotwarlost/istanbul/issues/677
- https://github.com/jestjs/jest/issues/4751

### PR DESCRIPTION
**Pull request by starnayuta**
Change the command for running eslint from node.js to npx for compatibility with Windows.

Closes #20

### CODE REVIEW

File: eslint-cjs-to-esm.js [hunk 1]
```diff
@@ -21,10 +18,10 @@ try {
         }
     });
     if (process.env.DEBUG === "eslint-cjs-to-esm") {
-        console.debug({ args, eslintBin, builtinConfig });
+        console.debug({ args, builtinConfig });
     }
-    const { stdout, stderr } = await execa("node", [
-        eslintBin,
+    const { stdout, stderr } = await execa("npx", [
```
Comment (azu) [inline eslint-cjs-to-esm.js:23]: I prefer to use `npm exec -- eslint <options>`
https://docs.npmjs.com/cli/v8/commands/npm-exec

**Comment by starnayuta** [comment]
The test results are as follows.
I could not determine if this result passed the test.

``` sh
PS C:\path\to\eslint-cjs-to-esm> node eslint-cjs-to-esm.js "./test-fixtures/*"  

C:\path\to\eslint-cjs-to-esm\test-fixtures\import-index.ts
  1:26  error  require file extension '/index.js'  file-extension-in-import-ts/file-extension-in-import-ts
C:\path\to\eslint-cjs-to-esm\test-fixtures\import-no-js-suffix.ts
  1:8  error  require file extension '.js'  file-extension-in-import-ts/file-extension-in-import-ts
  1:8  error  require file extension '.ts'  node/file-extension-in-import

C:\path\to\eslint-cjs-to-esm\test-fixtures\no-node-protocol.ts
  1:18  error  Prefer `node:path` over `path`  unicorn/prefer-node-protocol

C:\path\to\eslint-cjs-to-esm\test-fixtures\require.ts
  1:1  error  Expected "import" instead of "require()"  import/no-commonjs
  1:1  error  Do not use "require"                      unicorn/prefer-module

✖ 6 problems (6 errors, 0 warnings)
  4 errors and 0 warnings potentially fixable with the `--fix` option.

PS C:\path\to\eslint-cjs-to-esm> node eslint-cjs-to-esm.js "./test-fixtures/*" --fix

C:\path\to\eslint-cjs-to-esm\test-fixtures\require.ts
  1:1  error  Expected "import" instead of "require()"  import/no-commonjs
  1:1  error  Do not use "require"                      unicorn/prefer-module

✖ 2 problems (2 errors, 0 warnings)

```

**Review by azu — COMMENTED** [review]
https://github.com/azu/eslint-cjs-to-esm/blob/018695bdd7cc829062b7a1a4c0a784a51e5fed14/.github/workflows/test.yml#L5-L23

Can you add a matrix?

```
  test:
    name: Test(Node ${{ matrix.node }} on ${{ matrix.os }})
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest]
        node: [ 18, 20 ]
    name: Test(Node ${{ matrix.node }} on ${{ matrix.os }})
    runs-on: ${{ matrix.os }}

**Comment by azu** [comment]
https://github.com/azu/eslint-cjs-to-esm/blob/main/package.json#L40
This test confirms that the exit status is 1

**Comment by azu** [comment]
Another approach:

Simply, if process are Windows, just add `.cmd` like this.
https://github.com/saltyshiomix/resolve-as-bin/blob/23908b1c46827f6fd41cdcc401d3d2f5af76eade/index.ts#L13

**Comment by starnayuta** [comment]
Great. I will try this approach.

Thank you.

> Another option:
> 
> if process are Windows, just add `.cmd` like this. https://github.com/saltyshiomix/resolve-as-bin/blob/23908b1c46827f6fd41cdcc401d3d2f5af76eade/index.ts#L13

**Comment by starnayuta** [comment]
This error also happened on my local PC.
I am not sure what caused the error, but I will try to check.

https://github.com/azu/eslint-cjs-to-esm/actions/runs/7765836291/job/21181030640?pr=21

**Comment by azu** [comment]
https://github.com/azu/eslint-cjs-to-esm/actions/runs/7765836291/job/21181030640?pr=21
Ah, This may be expected-exit-status's issue.

**Comment by azu** [comment]
> ''node' is not recognized as an internal or external command,

Probably, need to use `node.exe` or `node.cmd` in Windows?

**Comment by starnayuta** [comment]
In cmd, single quote `'` was just a character.
It was changed to double quote `"`.

https://stackoverflow.com/questions/51080215/differences-between-single-and-double-quotes-in-cmd

**Comment by azu** [comment]
work on macOS/linux by following change:

``diff
- ./test-fixtures/*
+ ./test-fixtures/*.ts
```

**Comment by starnayuta** [comment]
Thank you for your kind assistance.

**Comment by azu** [comment]
Thanks for work!

Release it
https://github.com/azu/eslint-cjs-to-esm/releases/tag/v3.0.1 🎉

**Pull request merged**